src/store.ts

```diff
@@ -24,12 +24,13 @@ export const ingredients = derived(recipes, ($recipes) => {
     return ingredients;
 });
 
-export async function load_recipes() {
+export async function load_recipes(file:TFolder) {
```

Comment:
I believe the actual type here is a TAbstractFile

https://docs.obsidian.md/Reference/TypeScript+API/Vault/on.

Some pseudocode for making this a bit more intuitive. 

```
if file is folder and folder != recipe_folder return
if file is file and file.parent != recipe_folder return
```